### PR TITLE
add spec for $LOAD_PATH.resolve_feature_path

### PR DESCRIPTION
**Pull request by lxxxvi**
Hello 👋 

Here's my draft for `$LOAD_PATH.resolve_feature_path`.
I am quite certain that the specs are in the wrong place, can you tell me where to put it?

Thanks

### CODE REVIEW

**Comment by eregon** [comment]
> can you tell me where to put it?

It's a good question.
I think either `language/predefined_spec.rb` or `core/kernel/load_path_spec.rb` would be good places.
I have a slight preference for `language/predefined_spec.rb` since it's a really a singleton method of that global variable.

File: core/kernel/shared/require.rb [hunk 1]
```diff
@@ -774,3 +774,24 @@
...
+    it "returns what will be loaded without actual loading, .rb file" do
+      extension, path = $LOAD_PATH.resolve_feature_path('set')
+      extension.should == :rb
+      path.should.end_with?('/set.rb')
+    end
+
+    it "returns what will be loaded without actual loading, .so file" do
+      extension, path = $LOAD_PATH.resolve_feature_path('etc')
+      extension.should == :so
+      path.should =~ /\/etc\.(so|bundle)$/
```
Comment (eregon) [inline core/kernel/shared/require.rb:789]: Could you use `.end_with?("etc.#{RbConfig::CONFIG['DLEXT']}")`? (and add `require 'rbconfig` at the start of the `it`)

File: core/kernel/shared/require.rb [hunk 2]
```diff
@@ -774,3 +774,24 @@
...
+    end
+
+    # this may change (?!) https://bugs.ruby-lang.org/issues/16043
+    it "raises if feature cannot be found" do
```
Comment (eregon) [inline core/kernel/shared/require.rb:793]: Minor: `raises LoadError if ...`

**Comment by lxxxvi** [comment]
@eregon Thanks for your feedback.

I adjusted the specs according to your input and moved them to the bottom of `language/predefined_spec.rb`. 

Let me know if you have any additional input. 🙌

**Review by eregon — APPROVED** [review]
Thank you for the new specs, looks great!

**Pull request merged**